### ISSUE TITLE
Background task does not start

### ISSUE THREAD

**Issue by bluefly000** [ImranR98/Obtainium#1234]
**Describe the bug**

With the recent update, background tasks are no longer stuck while running, but instead, as in issue #87, background tasks are no longer started after the first few hours after opening Obtainium.
The interval is set to 1 hour and battery optimization is disabled.

**To Reproduce**

1. Open Obtainium.
2. Close Obtainium and wait a few hours.
3. Check the notification history.

**Screenshots and Logs**

Sorry for the Japanese screenshot. If you look at the time section, you will see that there are times when there are no notifications of background task running.

![1000001894_20240102210322](https://github.com/ImranR98/Obtainium/assets/101441707/682f2401-a286-4819-b7a9-e9eb20c325fa)

**Please complete the following information:**
 - Device: Google Pixel 6a, Pixel Tablet
 - OS: GrapheneOS
 - Obtainium Version: 0.15.3

**Comment by ImranR98** [comment]
The screenshot shows gaps of 4 to 10 hours when update checks weren't happening. Was the phone actively being used during those times? Did you relaunch Obtainium or did the checking start happening again on its own after those few hours?

**Comment by bluefly000** [comment]
The time in the screenshot from 3:00 to 7:00 was not rebooted and no other apps were used.
Also, between 9:00 and 19:00, the device was rebooted once for a system update (I think it was only rebooted in the evening).
I checked again with Pixel 6a and it seems to be working fine for the time between when I created this issue and now. I also checked by rebooting the Pixel Tablet and it seems that if the device is rebooted, the background task is not performed until Obtainium is opened once.
In any case, it seems that if for some reason a background task is not started once, it will not be done for the rest of the time.

**Comment by pxlkng** [comment]
I have a similar issue, but I am not sure if its related.

For me, after (re-)starting the phone, the Obtainium background update task **does not start at all** until I manually open the app.

If I, for example, reboot my phone, I wont get any notifications for new updates until I open Obtainium for the first time. Only then are all the new updates found and shown to me, and only after that I get automatic notifications for new updates going forward.

This also shows in the logs. It shows Obtainium checking all 3 hours for updates, until my phone auto rebooted (GOS setting). When I woke up, I unlocked the profile (BFU -> AFU) and used some apps, except Obtainium. A few more hours later (about 8 hours, so there should have happened multiple of the every-3-hour background checks) I checked on Obtainium and got greeted with an app update I hadn't gotten any notification for (the update was published even before I AFU-unlocked my phone). The logs didn't show any background update tasks in between me unlocking the profile and me checking on Obtainium.

But that's all I can observe. After having opened the app once, the task seems to run just fine for the whole time, until the phone gets rebooted.

Just to add that: I excluded Obtainium from background battery restrictions, so its set to Unrestricted.

**Phone:** Google Pixel 7a
**OS:** GrapheneOS (Build UQ1A.240105.004.2024010400, latest as of now)
**App version:** v0.15.10-beta
